starknet-providers/src/jsonrpc/mod.rs

```diff
@@ -634,7 +635,7 @@ where
                 JsonRpcMethod::GetStorageAt,
                 GetStorageAtRequestRef {
                     contract_address: contract_address.as_ref(),
-                    key: key.as_ref(),
+                    key: &StorageKey(format!("{:#x}", key.as_ref())),
```

Comment:
Technically, not all felts are valid storage keys, so typing key as `AsRef<FeltPrimitive>` is not valid. 

This opens up the user to failure in runtime. I think a correct approach would to be to change the signature and use a proper, validated struct for it. 
But since we have no validation for StorageKey, let's create an issue to address that in the future.